shared/chat/conversation/info-panel/add-to-channel/index.new.tsx

```diff
@@ -0,0 +1,201 @@
+  }
+
+  const loading = !allMembers.length
+
+  return (
+    <Kb.Modal
+      header={{
+        hideBorder: Styles.isMobile,
+        leftButton: Styles.isMobile ? (
+          <Kb.Text type="BodyBigLink" onClick={onClose}>
+            Cancel
+          </Kb.Text>
+        ) : (
+          undefined
+        ),
+        rightButton: Styles.isMobile && toAdd.size && (
+          <Kb.Text type="BodyBigLink" onClick={onAdd}>
+            Add
+          </Kb.Text>
+        ),
+        title: title({channelname, teamname}),
+      }}
+      footer={
+        Styles.isMobile
+          ? undefined
+          : {
+              content: (
+                <Kb.Box2 direction="horizontal" gap="tiny" fullWidth={true}>
+                  <Kb.Button
+                    type="Dim"
+                    label="Cancel"
+                    onClick={onClose}
+                    style={Styles.globalStyles.flexOne}
+                  />
+                  <Kb.Button
+                    label={toAdd.size ? `Add ${toAdd.size} ${pluralize('member', toAdd.size)}` : 'Add...'}
+                    onClick={onAdd}
+                    disabled={!toAdd.size}
+                    style={Styles.globalStyles.flexOne}
+                  />
+                </Kb.Box2>
+              ),
+            }
+      }
+      onClose={onClose}
+      allowOverflow={true}
+    >
+      <Kb.SearchFilter
+        onChange={text => setFilter(text)}
+        size="full-width"
+        placeholderText={
+          loading ? 'Loading...' : `Search ${allMembers.length} ${pluralize('member', allMembers.length)}`
+        }
+        style={styles.filterInput}
+      />
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.listContainer}>
+        <Kb.List2
+          items={membersFiltered}
+          renderItem={(idx, item) => {
+            const alreadyIn = participants.includes(item.username)
+            const onCheck = () => {
+              if (toAdd.has(item.username)) {
+                toAdd.delete(item.username)
+                setToAdd(new Set(toAdd))
+              } else {
+                toAdd.add(item.username)
+                setToAdd(new Set(toAdd))
+              }
+            }
+            return (
+              <Kb.ListItem2
+                firstItem={!Styles.isMobile || idx === 0}
+                icon={<Kb.Avatar size={32} username={item.username} />}
+                type="Small"
+                onClick={alreadyIn ? undefined : onCheck}
+                body={
+                  <Kb.Box2 direction="vertical" alignItems="flex-start">
+                    <Kb.ConnectedUsernames
+                      type="BodySemibold"
+                      colorFollowing={true}
+                      usernames={[item.username]}
+                    />
+                    <Kb.Text type="BodySmall" lineClamp={1}>
+                      {alreadyIn && <Kb.Text type="BodySmall">Already in{!!item.fullName && ' • '}</Kb.Text>}
+                      {item.fullName}
```

Comment:
Design shows name first, "Already in" after.
![image](https://user-images.githubusercontent.com/59594/74200152-ffa3b180-4c1a-11ea-88ea-ad77251c5c21.png)

Reply:
Also, is it all right to have naked text next to a `Text` tag?

Reply:
Yeah, I made that change to deal with long fullnames. I was going for:
```
a very very long fullna... • Already in
```
But the name kept pushing "Already in" off the side.

Reply:
`{item.fullName}` is still inside the outer `Text`